File: backend/src/test/java/lan/dk/podcastserver/entity/StatusTest.java
package lan.dk.podcastserver.entity;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Created by kevin on 14/06/15 for HackerRank problem
 */
public class StatusTest {

    @Test
    public void should_check_value() {
        assertThat(Status.of("NOT_DOWNLOADED")).isEqualTo(Status.NOT_DOWNLOADED);
    }
    
    @Test(expected = IllegalArgumentException.class)
    public void should_throw_exception() {
        assertThat(Status.of("")).isNull();
    }
}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/entity/StatusTest.kt
package com.github.davinkevin.podcastserver.entity

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

/**
 * Created by kevin on 14/06/15 for HackerRank problem
 */
class StatusTest {

    @Test
    fun `should check value`() {
        assertThat(Status.of("NOT_DOWNLOADED"))
                .isEqualTo(Status.NOT_DOWNLOADED)
    }

    @Test
    fun `should throw exception`() {
        assertThatThrownBy { Status.of("") }
                .isInstanceOf(IllegalArgumentException::class.java)
                .hasMessage("No enum constant Status.")
    }
}
